sources/osgShadow/ShadowTextureAtlas.js

```diff
@@ -0,0 +1,199 @@
+
+        this._lightUnit = lun;
+
+    },
+    getLightUnit: function () {
+
+        return this._lightUnit;
+
+    },
+
+    getUniformName: function ( name, lightNumber ) {
+
+        var prefix = 'Shadow_' + this.getType() + this._lightUnit.toString();
+        if ( lightNumber !== undefined ) prefix += '_' + lightNumber;
+        return 'u' + prefix + '_' + name;
+
+    },
+
+    getVaryingName: function ( name, lightNumber ) {
+
+        var prefix = this.getType() + ( this._lightUnit + lightNumber ).toString();
+        return 'v' + prefix + '_' + name;
+
+    },
+
+    getOrCreateUniforms: function ( unit ) {
+
+        // uniform are once per CLASS attribute, not per instance
+        var obj = ShadowTextureAtlas;
+        console.assert( unit !== undefined );
+        Notify.assert( this._lightUnit !== -1 );
+
+        if ( obj.uniforms[ unit ] !== undefined ) {
+            return obj.uniforms[ unit ];
+        }
+
+        var uniforms = {};
+
+
+        // shadowmap texture size used for texel space which is viewport independant
+        var renderSizeUniform = Uniform.createFloat4( this.getUniformName( 'renderSize' ) );
+        uniforms[ 'RenderSize' ] = renderSizeUniform;
+
+        for ( var i = 0; i < this._lightRange; i++ ) {
+
+            uniforms[ 'ViewMatrix_' + i ] = Uniform.createMat4( this.getUniformName( 'viewMatrix', i ) );
+            uniforms[ 'ProjectionMatrix_' + i ] = Uniform.createMat4( this.getUniformName( 'projectionMatrix', i ) );
+            uniforms[ 'DepthRange_' + i ] = Uniform.createFloat4( this.getUniformName( 'depthRange', i ) );
+            uniforms[ 'MapSize_' + i ] = Uniform.createFloat4( this.getUniformName( 'mapSize', i ) );
+            uniforms[ 'RenderSize_' + i ] = renderSizeUniform;
+
+        }
+
+
+        // Dual Uniform of texture, needs:
+        // - Sampler (type of texture)
+        // - Int (texture unit)
+        // tells Shader Program where to find it
+        var name = 'Texture' + unit;
+        var uniform = Uniform.createInt1( unit, name );
+        uniforms[ name ] = uniform;
+
+        // Per Class Uniform Cache
+        obj.uniforms[ unit ] = new Map( uniforms );
+
+        return obj.uniforms[ unit ];
+    },
+
+    setViewMatrix: function ( viewMatrix, lightNumber ) {
+        this._viewMatrices[ lightNumber ] = viewMatrix;
+    },
+
+    setProjectionMatrix: function ( projectionMatrix, lightNumber ) {
+        this._projectionMatrices[ lightNumber ] = projectionMatrix;
+    },
+
+    setDepthRange: function ( depthRange, lightNumber ) {
+        this._depthRanges[ lightNumber ] = depthRange;
+    },
+
+    setTextureSize: function ( w, h ) {
+
+        this._renderSize[ 0 ] = w;
+        this._renderSize[ 1 ] = h;
+        this._renderSize[ 2 ] = 1.0 / w;
+        this._renderSize[ 3 ] = 1.0 / h;
+        Texture.prototype.setTextureSize.call( this, w, h );
+        this.dirty();
+
+    },
+
+    setMapSize: function ( dimension, lightNumber ) {
```

Comment:
I would expect lightNumber first argument, (lightNumber, dimension) because it's an important parameter. same thing for other call that needs lightNumber as argument